drivers/applications/tla2528_analog_input_demo.cpp

```diff
@@ -0,0 +1,39 @@
+#include "../hardware_map.hpp"
+#include <array>
+#include <libhal-util/bit.hpp>
+#include <libhal-util/i2c.hpp>
+#include <libhal-util/serial.hpp>
+#include <libhal-util/steady_clock.hpp>
+#include <libhal/input_pin.hpp>
+#include <libhal/units.hpp>
+#include <tla2528.hpp>
+#include <tla2528_adapters.hpp>
+
+using namespace hal::literals;
+using namespace std::chrono_literals;
+
+namespace sjsu::drivers {
+
+void application(application_framework& p_framework)
+{
+
+  hal::byte i2c_address = 0x10;  // 0x10 is the tla address for no resistors
+
+  auto& terminal = *p_framework.terminal;
+  auto& i2c = *p_framework.i2c;
+  auto& steady_clock = *p_framework.steady_clock;
```

Comment:
```suggestion
  auto& terminal = *p_framework.terminal.value();
  auto& i2c = *p_framework.i2c.value();
  auto& steady_clock = *p_framework.steady_clock.value();
```

That way it will throw an exception if the value is empty.

Reply:
for me if I put  `.value()` the IED doesn't recognize it and if I try to build it spits out an error

Reply:
I see. I added this technical debt item #14 so we can track a move to `std::optional` for this.